partiql-plan/src/main/java/org/partiql/plan/rel/RelCorrelate.java

```diff
@@ -16,6 +16,10 @@
 public abstract class RelCorrelate extends RelBase {
```

Comment:
One thing to check before releasing is that RelCorrelate is actually used for lateral joins. Can you check this?

Reply:
discussed offline: `RelCorrelate` is based off of Calcite's `Correlate` node https://calcite.apache.org/javadocAggregate/org/apache/calcite/rel/core/Correlate.html. It's currently unused so for now just marked as experimental.

In the future, we may add it back and have `RelJoin` correspond to the de-correlated join and have `RelCorrelate` correspond to correlated joins.